content/2. Ondersteunende-informatie/4. Techniek/Git/8. Git files.md

```diff
@@ -0,0 +1,49 @@
+---
+title: 8. Git Files
+taxonomie:
+  - bg-24.2.Git-Files.OI
+  - bg-24.3.Git-Files.OI
+---
+
+Met Git zijn er sommige documenten of folders wat Git herkent en kan gebruiken voor eigen acties. 
+
+## .git/
+De `.git` folder is een verborgen map in de root van een project. De map bevat alle metadata en objecten die Git nodig heeft om de repository te herkennen en beheren. Hierbij kan je denken aan de commitgeschiedenis, configuraties, branches, tags en stashes. Verder zijn er ook nog folders in de `.git` folder die bepaalde taken hebben
+- `objects/`: Bevat alle objecten die Git nodig heeft voor het versiebeheer (zoals commits).
+- `refs/`: Bevat verwijzingen naar branches en tags.
+- `config`: De repository-specifieke configuratie.
+- `HEAD`: Bevat een verwijzing naar de huidige actieve branch.
+
+### .git/hooks/
+De `hooks` subfolder bevat scripts die automatisch worden uitgevoerd door Git bij bepaalde gebeurtenissen of acties in de levenscyclus van een repository. Deze gebeurtenissen omvatten zaken zoals commits, merges, pushes, en meer. 
+
+[Git hooks](https://git-scm.com/book/ms/v2/Customizing-Git-Git-Hooks) geven je de mogelijkheid om taken te automatiseren, zoals het valideren van code voordat je deze commit, het versturen van meldingen na een push, of het controleren van codeformattering. Een paar voorbeelden van zo'n hook is een `pre-commit`, `post-commit` en `pre-push`. 
```

Comment:
Opmerking: dit klinkt als erg handig spul voor projecten! Komt daar later (bij het procedurele deel) een voorbeeld van? Ik zet de vraag hier alvast neer, voor het geval ik het zelf niet terugzie.

Reply:
Op het moment is dit nog geen onderdeel van de PI, maar kan later altijd nog worden toegevoegd.